File: daily/Linked List/3217-Delete-nodes-from-linked-list-present-in-array.py
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        seen = set()
        for num in nums:
            seen.add(num)
        
        dummy = ListNode(0)
        curr = dummy
        while head:
            if head.val not in seen:
                curr.next = head
                curr = curr.next
            head = head.next
        curr.next = None
        return dummy.next

File: daily/Linked List/3217-Delete-nodes-from-linked-list-present-in-array.go
package daily

func modifiedList(nums []int, head *ListNode) *ListNode {
	seen := make(map[int]bool)
	for _, num := range nums {
		seen[num] = true
	}

	dummy := &ListNode{Val: 0}
	curr := dummy
	for head != nil {
		if !seen[head.Val] {
			curr.Next = head
			curr = curr.Next
		}
		head = head.Next
	}
	curr.Next = nil
	return dummy.Next
}
